src/com/andrewkingmarshall/Utils/PlaylistHandler.java

```diff
@@ -45,12 +45,21 @@ public void joinPlaylist(SpoqUser spoqUser) throws PlaylistNotFoundException{
         if(playlistExists(spoqUser.getConnectedPlaylistId())) {
             spoqUser.setAdmin(false);
             playlistMap.get(spoqUser.getConnectedPlaylistId()).addUser(spoqUser);
+        }else{
+            throw new PlaylistNotFoundException();
         }
     }
 
     public void leavePlaylist(SpoqUser spoqUser) throws PlaylistNotFoundException{
-        if(playlistExists(spoqUser.getConnectedPlaylistId())) {
-            playlistMap.get(spoqUser.getConnectedPlaylistId()).removeUser(spoqUser.getMusicServiceId());
+        int playlistId = spoqUser.getConnectedPlaylistId();
+        System.out.println("Leaving playlist: " + playlistId);
```

Comment:
My reasoning for this is so that "Leaving" signals we are currently attempting to leave the playlist but when the user is actually removes we signal that with "Removing user". I agree that it could be confusing though, so perhaps we change "leaving" to "attemping to leave" and also add that else statement?